Library/Homebrew/formula.rb

```diff
@@ -1843,7 +1843,12 @@ def eligible_kegs_for_cleanup
       eligible_kegs = if head? && (head_prefix = latest_head_prefix)
         installed_kegs - [Keg.new(head_prefix)]
       else
-        installed_kegs.select { |k| pkg_version > k.version }
+        installed_kegs.select do |keg|
+          tab = Tab.for_keg(keg)
```

Comment:
That's the default but I was struggling to see where it's set to `0` if it's not present in an (old) tab? Happy to defer to you here though if you're sure.